### PR TITLE
Store blockheaders pushed from Agora to STOA

### LINKED ISSUE #386: Store blockheaders pushed from Agora to STOA
Block signature information in the blockhead can be updated.
This can be renewed even after the block is externalized.

Reference https://github.com/bosagora/agora/pull/2234

### PR DESCRIPTION
**Pull request by TrustHenry**
POST `/block_header_updated` The endpoint has been added.
Puts the history of the changes blockheader in a separate table.

### CODE REVIEW

File: docs/Database_Schema.md [hunk 1]
```diff
@@ -279,7 +279,35 @@ CREATE TABLE IF NOT EXISTS "merkle_trees" (
...
+| signature         | TINYBLOB  |    | Y        |          | Schnorr multisig of all validators which signed this block |
+| missing_validators| TEXT      |    | N        |          | List of indices to the validator UTXO set which have not revealed the preimage |
+
+### _Create Script_
+
+```sql
+CREATE TABLE IF NOT EXISTS "blocks_header_updated_history" 
+    "block_height"        INTEGER  NOT NULL,
+    "updated_time"        INTEGER  NOT NULL,
+    "hash"                TINYBLOB NOT NULL,
```
Comment (MichaelKim20) [inline docs/Database_Schema.md:303]: We need updated_height. 
This is the block height when the blockhead is updated.

File: tests/Storage.test.ts [hunk 2]
```diff
@@ -12,9 +12,15 @@
 *******************************************************************************/
 
 import * as assert from "assert";
-import { Block, Endian, Hash, Height, JSBI, PreImageInfo, SodiumHelper } from "boa-sdk-ts";
+import {Block, BlockHeader, Endian, Hash, Height, JSBI, PreImageInfo, SodiumHelper} from "boa-sdk-ts";
```
Comment (MichaelKim20) [inline tests/Storage.test.ts]: Isn't the coding style automated? Please check it.

**Comment by MichaelKim20** [comment]
Shouldn't we add it to blockheader's history table when every block is recorded?

File: docs/Database_Schema.md (hunk 1 above)
Comment (TrustHenry) [reply docs/Database_Schema.md:303]: I will add `current_height`

File: tests/Storage.test.ts (hunk 2 above)
Comment (TrustHenry) [reply tests/Storage.test.ts]: I Applied it.

**Comment by TrustHenry** [comment]
> Shouldn't we add it to blockheader's history table when every block is recorded?

**Comment by TrustHenry** [comment]
Thank you for your reviews. I apply your suggestions.

**Review by MichaelKim20 — APPROVED** [review]

**Review by kchulj — APPROVED** [review]

**Pull request merged**